### PR TITLE
R Processing syntax summary table

### PR DESCRIPTION
**Pull request by ghtmtt**
New chapter in the Training manual with tables that summarize the input and output option in R Processing syntax style

### CODE REVIEW

File: source/docs/training_manual/processing/r_syntax-table.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+| point          | P = point                        | when clicked on the map, you have the coordinates of the point                |
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+| file           | F = file                         | path of the file chosen, e.g. "/home/matteo/file.txt"                         |
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+| folder         | F = folder                       | path of the folder chosen, e.g. "/home/matteo/Downloads"                      |
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+
+Any of the input could be also **OPTIONAL**, that is that if you have a handy way to tell the script to ignore this parameter.
```
Comment (DelazJ) [inline source/docs/training_manual/processing/r_syntax-table.rst]: `that is that if you have a handy way to tell the script to ignore this parameter.` 
Isn't there something wrong with this part?

**Comment by ghtmtt** [comment]
mm I don't think so.. I meant, if you add **optional** to the input parameter, than the algorithm just ignores it.. should I provide a better explanation?

**Comment by yjacolin** [comment]
@ghtmtt do you mean: `Any of the input could be also OPTIONAL, that means that you have a handy way to tell the script to ignore this parameter.`

**Comment by ghtmtt** [comment]
exactly! I will change that

**Comment by ghtmtt** [comment]
Hi guys.. I've just added the contribution header at the top of the 3 files of the PR.. no big changes.. any chance to have also the last file (r-syntax-table) merged? Is it ok?

**Comment by yjacolin** [comment]
LGTM for me. Just wait for the test (I stoped the 2 before the last one as they are unuseful).

File: source/docs/training_manual/processing/r_syntax-table.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+| point          | P = point                        | when clicked on the map, you have the coordinates of the point                |
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+| file           | F = file                         | path of the file chosen, e.g. "/home/matteo/file.txt"                         |
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+| folder         | F = folder                       | path of the folder chosen, e.g. "/home/matteo/Downloads"                      |
++----------------+----------------------------------+-------------------------------------------------------------------------------+
+
+Any of the input could be also **OPTIONAL**, that means that you have a handy way to tell the script to ignore this paramete
```
Comment (DelazJ) [inline source/docs/training_manual/processing/r_syntax-table.rst]: paramete**r**.

File: source/docs/training_manual/processing/r_syntax-table.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+In order to set an input as optional, you just have to add the string ``optional`` **before** the input, e.g::
+
+  ##Layer = vector
+  ##Field1 = Field Layer
+  ##Field2 = optional Field Layer
+
+
+Output parameters
+-----------------
+Output parameters takes the **Input** names you gave at the beginning of the script and write the object you want.
```
Comment (DelazJ) [inline source/docs/training_manual/processing/r_syntax-table.rst]: take~~**s**~~?

**Comment by DelazJ** [comment]
Found some typos. And still need to split lines to 80 characters.
Otherwise, I don't have skills to discuss the content but looks good (for the structure)

**Comment by ghtmtt** [comment]
typos corrected.. for the line length.. I used an online rst editor to be sure that the tables look good.. 
http://rst.ninjs.org/

you can just copy and paste the code and have a look if something is missing and the table crashes

thanks!

**Comment by DelazJ** [comment]
I was refering to description text, not table. But never mind. No problem to merge!

**Comment by ghtmtt** [comment]
OK thanks!

**Pull request merged**